samples/drivers/kscan_touch/src/main.c

```diff
@@ -19,6 +19,12 @@ LOG_MODULE_REGISTER(main);
 
 const struct device *kscan_dev = DEVICE_DT_GET(TOUCH_CONTROLLER_NODE);
 
+#ifdef CONFIG_BOARD_STM32L562E_DK
```

Comment:
Introducing board-specific hacks into generic samples is a no-no.

I suggest you do something like add some optional /zephyr,user properties that configure whatever quirks you need to manage here instead.

Reply:
I agree with that, I don't like to have enable the LCD/touchscreen power in the kscan touch sample.
Normally, it is enabled by the display driver, but since the power lines are shared between the touch and the lcd controller, I'm a bit unsure on how to handle this.

I put the priority on the display driver since its configuration is relying on powering up the display and waiting enough time. 
Overwise, the only thing I see is that touch kscan driver needs a new device-tree property to also add an optionnal power-gpio to it and enable it in the init routine of the kscan device. Then we'll have to handle the fact that the power-gpio will be manipulated by both the display and touch driver at runtime.

Reply:
Fixed in latest commit. I also updated the FT5336 driver to the latest GPIO_DT api. There are no more traces of the board specific changes in the kscan driver.